### PR TITLE
Implemented Binary tree

### LINKED ISSUE #6: Binary Tree
Implement a complete functioning Binary Tree

### PR DESCRIPTION
**Pull request by 1c3t3a**
Closes #6. Just misses a remove method, but this could be done fixing a several Issue (#21).

### CODE REVIEW

**Comment by 1c3t3a** [comment]
# [Codecov](https://codecov.io/gh/1c3t3a/rust-data-structures/pull/22?src=pr&el=h1) Report
> :exclamation: No coverage uploaded for pull request base (`master@ab74686`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference#section-missing-base-commit).
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/1c3t3a/rust-data-structures/pull/22?src=pr&el=tree)

```diff
@@            Coverage Diff            @@
##             master      #22   +/-   ##
=========================================
  Coverage          ?   93.77%           
=========================================
  Files             ?        3           
  Lines             ?      562           
  Branches          ?        0           
=========================================
  Hits              ?      527           
  Misses            ?       35           
  Partials          ?        0           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/1c3t3a/rust-data-structures/pull/22?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/1c3t3a/rust-data-structures/pull/22?src=pr&el=footer). Last update [ab74686...0565d1b](https://codecov.io/gh/1c3t3a/rust-data-structures/pull/22?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: src/binary_tree.rs [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+    /// super unordered.
+    /// # Example
+    /// ```rust
+    /// use data_structure_with_colin::binary_tree::BinaryTree;
+    /// let mut binary_tree = BinaryTree::new();
+    /// binary_tree.insert(1);
+    ///
+    /// assert!(binary_tree.find(1));
+    /// ```
+    pub fn find(&self, value: T) -> bool {
```
Comment (Colin1860) [inline src/binary_tree.rs:81]: pls rename to contains, find implies you return the value (or do that and keep the name)

File: src/binary_tree.rs [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+            return None;
+        }
+
+        let node = self.visited.pop();
+        match (
+            node.unwrap().as_ref().left.as_ref(),
+            node.unwrap().as_ref().right.as_ref(),
+        ) {
+            (None, None) => (),
+            (leaf @ Some(_), None) | (None, leaf @ Some(_)) => self.visited.push(&leaf.unwrap()),
```
Comment (Colin1860) [inline src/binary_tree.rs:299]: Magic!!

**Review by Colin1860 — CHANGES_REQUESTED** [review]

File: src/binary_tree.rs (hunk 2 above)
Comment (Colin1860) [reply src/binary_tree.rs:299]: You can resolve this, comment i just wanted you to see it 🥇

Comment (1c3t3a) [reply src/binary_tree.rs:299]: 😂😂😂

**Review by Colin1860 — APPROVED** [review]

**Pull request merged**